### ISSUE TITLE
Editor is not accessible to screen readers

### ISSUE THREAD

**Issue by toolness** [mozilla/thimble.mozilla.org#1470]
I noticed that when using [NVDA](http://www.nvaccess.org/) that the editor (a CodeMirror widget, it seems) is not particularly accessible. Moving the cursor seems to indicate that every single line is "blank".

I did a bit of research on this and found that the CodeMirror widget is not accessible by default on desktop platforms, but [manually setting the `inputStyle` option](http://codemirror.net/doc/manual.html#option_inputStyle) when creating a CodeMirror instance *hugely* improves this. The main problem is that older browsers need to use the non-accessible default to work well for sighted users, but there are a variety of ways to potentially work around this.

The CodeMirror documentation I linked to above does mention that "The intention is to make [the accessible option] the default on modern desktop browsers in the future.", and I'm curious what it means by "modern browsers". For example, if Thimble *already* requires a "modern browser" to use, might it be acceptable to simply force the accessible option on at all times?

**Comment by Pomax** [comment]
Nice discovery! I wonder how easily we can set that option given that this is part of Brackets... @gideonthomas, @humphd any idea?

**Comment by toolness** [comment]
Hi Pomax!

Here's a page that contains two codemirror widgets, one which sets `inputStyle: "contenteditable"` and the other which sets `inputStyle: "textarea"`:

http://jsbin.com/qopaxex/edit?html,output

Hopefully folks interested in accessibility can use that to see the difference between the two when used with screen-readers.

**Comment by humphd** [comment]
Brackets sets the CodeMirror instance here: https://github.com/adobe/brackets/blob/656a0ee1d9889e853a77a4d9b12b9a82e6f86c55/src/editor/Editor.js#L389

We can hack it, sure.
